horizons/__init__.py

```diff
@@ -0,0 +1,281 @@
+               'uranus': '799', 'neptune': '899', 'pluto': '999', 'io': '501'}
+
+    def __init__(self,
+                 site: Site,
+                 path: str = 'horizons/data',
+                 start: datetime = None,
+                 end: datetime = None):
+        self.path = path
+        self.url = 'https://ssd.jpl.nasa.gov/horizons_batch.cgi'
+        self.start = start
+        self.end = end
+        self.site = site
+     
+    
+    @staticmethod
+    def generate_horizons_id(designation: str) -> str:
+        des = designation.lower()
+        return HorizonsClient.bodies[des] if des in HorizonsClient.bodies else designation
+    
+    @staticmethod
+    def _angular_distance(ra1: float, dec1: float, ra2: float, dec2: float) -> float:
+        """
+        Calculate the angular distance between two points on the sky.
+        based on https://github.com/gemini-hlsw/lucuma-core/blob/master/modules/core/shared/src/main/scala/lucuma/core/math/Coordinates.scala#L52
+        """
+        φ1 = dec1
+        φ2 = dec2
+        delta_φ = dec2 - dec1
+        delta_λ = ra2 - ra1
+        a = np.sin(delta_φ / 2)**2 + np.cos(φ1) * np.cos(φ2) * np.sin(delta_λ / 2)**2
+        return 2 * np.arctan2(np.sqrt(a), np.sqrt(1 - a))
+
+    @staticmethod
+    def interpolate(ra1: float, dec1: float, ra2: float, dec2: float, f: float) -> float:
+        """
+        Interpolate between two coordinates.
+        based on https://github.com/gemini-hlsw/lucuma-core/blob/master/modules/core/shared/src/main/scala/lucuma/core/math/Coordinates.scala#L63-L89
+        """
+        # calculate angular distance to radians
+        delta = HorizonsClient.angular_distance(ra1, dec1, ra2, dec2)
+        if delta == 0:
+            return ra1, dec1
+        else:
+
+            φ1 = dec1
+            φ2 = dec2
+            λ1 = ra1
+            λ2 = ra2
+            a = np.sin((1 - f) * delta) / np.sin(delta)
+            b = np.sin(f * delta) / np.sin(delta)
+            x = a * np.cos(φ1) * np.cos(λ1) + b * np.cos(φ2) * np.cos(λ2)
+            y = a * np.cos(φ1) * np.sin(λ1) + b * np.cos(φ2) * np.sin(λ2)
+            z = a * np.sin(φ1) + b * np.sin(φ2)
+            φi = np.arctan2(z, np.sqrt(x * x + y * y))
+            λi = np.arctan2(y, x)
+            return λi, φi # there is a transformation here
+
+    def _form_horizons_name(self, tag: TargetTag, designation: str) -> str:
+        """
+        Formats the name of the body
+        """
+        if tag is TargetTag.COMET:
+            name = f'DES={designation};CAP'
+        elif tag is TargetTag.ASTEROID:
+            name = f'DES={designation};'
+        else:
+            name = self.generate_horizons_id(designation)
+        return name
+
+    def _time_bounds(self) -> Tuple[str, str]:
+        """
+        Returns the start and end times based on the given date
+        """
+        return self.start.strftime('%Y%m%d_%H%M'), self.end.strftime('%Y%m%d_%H%M')
```

Comment:
We should probably extract `%Y%m%d_%H%M` as a static member of this class since we're using it twice.